frame/staking/src/voter_bags.rs

```diff
@@ -458,19 +459,56 @@ impl<T: Config> Bag<T> {
 	/// the first place. Generally, use [`VoterList::remove`] instead.
 	///
 	/// Storage note: this modifies storage, but only for adjacent nodes. You still need to call
-	/// `self.put()` and `node.put()` after use.
+	/// `self.put()`, `VoterNodes::remove(voter_id)` and `VoterBagFor::remove(voter_id)`
+	/// to update storage for the bag and `node`.
 	fn remove_node(&mut self, node: &Node<T>) {
-		// TODO: we could merge this function here.
-		// node.excise();
+		// Excise `node`.
 		if let Some(mut prev) = node.prev() {
-			prev.next = self.next.clone();
+			debug_assert!(
+				self.head.as_ref() != Some(&node.voter.id),
+				"node is the head, but has Some prev"
+			);
+			debug_assert!(
+				prev.prev().is_some() || self.head.as_ref() == Some(&prev.voter.id),
+				"node.prev.prev should be Some OR node.prev should be the head"
+			);
```

Comment:
What if I ask you to check that `pre.prev().prev().is_some()` is `Some` or tail? 

I think what we should do is to write a 

```
fn sanity_check_voter_list() {
	// iterate all voters and ensure only those that are head don't have a prev
    // iterate all voters and ensure only those that are tail don't have a next
    // ensure counters are always sane (VoterCount == CounterForValidators. + CounterForNominators)
    // anything else..
}
```

and sprinkle this as debug assertions after each mutating operation on the public interface of the bags module. 

Probably we can retire some existing ones in favour of this.